### DISCUSSION TITLE
search similar vectors from mysql with sql query

### DISCUSSION THREAD

**Discussion by sadakmed** [milvus-io/milvus#6101] [Q&A and General discussion]
I'm investigating the possibility to store embeddings, and then search for similar embeddings giving an id from mysql(using sql queries)

Is it possible?

**Comment by yhmo** [comment]
Two years ago, when we begin to design a vector search engine, we developed a prototype on PostgreSQL. The PG allow you store float array (vector data), and we inject user defined function into PostgreSQL to calculate vector distance. But the performance is poor, because the pg internal machinery is not suitable to parallel computing for vector search. I think it is possible to build another prototype on MySQL, but may face lot of limitation.

**Reply by sadakmed** [reply]
what about saving the distance between all vectors nd store that as a metadata in one of the tables?
nd if a new vector is introduced, the calculation will target only that vector and the existing(nd not between the existing ones **again**)

**Reply by yhmo** [reply]
Do you mean:
add No.1 vector, do nothing
add No.2 vector, calculate 1 distance
add No.3 vector, calculate 2 distance
......
add No.n vector, calculate n-1 distance

When n is a large value, we say, 100 million, then each time you add a new vector, it need to calculate 100M-1 distance. The performance must be poor.
